test/core/compression/message_compress_test.cc

```diff
@@ -196,6 +196,34 @@ static void test_bad_decompression_data_crc(void) {
   grpc_slice_buffer_destroy(&output);
 }
 
+static void test_bad_decompression_data_missing_trailer(void) {
+  grpc_slice_buffer input;
+  grpc_slice_buffer corrupted;
+  grpc_slice_buffer output;
+  size_t idx;
+
+  grpc_slice_buffer_init(&input);
+  grpc_slice_buffer_init(&corrupted);
+  grpc_slice_buffer_init(&output);
+  grpc_slice_buffer_add(&input, create_test_value(ONE_MB_A));
+
+  grpc_core::ExecCtx exec_ctx;
+  /* compress it */
+  grpc_msg_compress(GRPC_MESSAGE_COMPRESS_GZIP, &input, &corrupted);
+  /* corrupt the output by smashing the CRC */
+  GPR_ASSERT(corrupted.count > 1);
+  GPR_ASSERT(GRPC_SLICE_LENGTH(corrupted.slices[1]) > 8);
```

Comment:
Are you sure the compressed slice buffer will always contain 2 slices? Maybe worth some more sophisticated approach to remove the last 8 bytes of the buffer.